EventFilter/HcalRawToDigi/plugins/PackerHelp.h

```diff
@@ -401,14 +418,28 @@ class UHTRpacker {
 
     int fiber = eid.fiberIndex();
     int fiberchan = eid.fiberChanId();
-    int capid0 = qiedf.samples() == 0 ? 0 : qiedf[0].capid();  // capacitor id for the first sample
-
-    header |= (fiberchan & QIE11HeaderSpec::MASK_FIBERCHAN) << QIE11HeaderSpec::OFFSET_FIBERCHAN;
-    header |= (fiber & QIE11HeaderSpec::MASK_FIBER) << QIE11HeaderSpec::OFFSET_FIBER;
-    header |= (capid0 & QIE11HeaderSpec::MASK_CAPID) << QIE11HeaderSpec::OFFSET_CAPID;
-    header |= (0x0 & QIE11HeaderSpec::MASK_FIBERERR) << QIE11HeaderSpec::OFFSET_FIBERERR;
-    header |= (0x0 & QIE11HeaderSpec::MASK_FLAVOR) << QIE11HeaderSpec::OFFSET_FLAVOR;  //flavor
-    header |= (0x1 & QIE11HeaderSpec::MASK_HEADER_BIT) << QIE11HeaderSpec::OFFSET_HEADER_BIT;
+    // capacitor id for the first sample
+
+    int flavor = qiedf[0].flavor();
+    int capid0;
+    if (qiedf.samples() == 0)
+      capid0 = 0;
+    else if (flavor == 3) {
+      header |= (fiberchan & QIE11HeaderSpec3::MASK_FIBERCHAN) << QIE11HeaderSpec3::OFFSET_FIBERCHAN;
```

Comment:
cupid0 assignment forgotten here

Reply:
This is intended, since capid is not stored in the header of HB.

Reply:
Uhm...  Before commit https://github.com/cms-sw/cmssw/pull/28330/commits/768d39c706ed0c293ec0b0337b94112631730e2d ```capid0``` was defined also in case of ```flavor == 3```, see previous L430 of EventFilter/HcalRawToDigi/plugins/PackerHelp.h. Was it a bug, then?

In view of this latest statement, then you could simplify the code (and not remain with unassigned variables in it). In EventFilter/HcalRawToDigi/plugins/PackerHelp.h please:
- Move the definitions of ```int capid0``` and ```int fiberErr``` at L345-355 inside the ```if``` clause after the ```else``` at L361
- Remove the useless ```if (qiedf.samples() == 0)``` part at L425-426m and move the definition of ```capid0``` together with the assignment at L435

Reply:
yeah capid0 is declared a bit early but only needs to be used for flavor==0 case. changes made according to suggestion for simplification.